### PR TITLE
Feature implement user authentication with passport

### PR DESCRIPTION
**Pull request by ahmadmhm**
review

### CODE REVIEW

File: database/migrations/2014_10_12_000000_create_users_table.php [hunk 1]
```diff
@@ -15,12 +15,16 @@ public function up()
...
             $table->id();
-            $table->string('name');
+            $table->string('fullname');
             $table->string('email')->unique();
+            $table->string('mobile')->unique();
             $table->timestamp('email_verified_at')->nullable();
-            $table->string('password');
+            $table->timestamp('mobile_verified_at')->nullable();
+            $table->string('password')->nullable();
+            $table->enum('status', ['active', 'inactive', 'pending', 'block'])->nullable();
```
Comment (ahmadmhm) [inline database/migrations/2014_10_12_000000_create_users_table.php]: It should not to be nullable, please remove that

File: app/Models/User.php [hunk 2]
```diff
@@ -3,24 +3,25 @@
 namespace App\Models;
 
 // use Illuminate\Contracts\Auth\MustVerifyEmail;
```
Comment (ahmadmhm) [inline app/Models/User.php]: Please, Remove comment lines

File: app/Models/User.php [hunk 3]
```diff
@@ -3,24 +3,25 @@
...
 {
-    use HasApiTokens, HasFactory, Notifiable;
+    use HasApiTokens, HasFactory;
 
     /**
      * The attributes that are mass assignable.
      *
      * @var array<int, string>
      */
     protected $fillable = [
```
Comment (ahmadmhm) [inline app/Models/User.php]: This changes must be in the user migration commit!!!

File: database/migrations/2014_10_12_000000_create_users_table.php (hunk 1 above)
Comment (Niramkin) [reply database/migrations/2014_10_12_000000_create_users_table.php]: line 24 nullable was removed and defult was added

**Pull request closed without merging**